mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+    input data as a Spark DataFrame prior to scoring. Also supports deployment in Spark
+    as a Spark UDF. Models with this flavor can be loaded as Python functions
+    for performing inference. This flavor is always produced.
+:py:mod:`mlflow.mleap`
+    Enables high-performance deployment outside of Spark by leveraging MLeap's
+    custom dataframe and pipeline representations. Models with this flavor *cannot* be loaded
+    back as Python objects. Rather, they must be deserialized in Java using the
+    ``mlflow/java`` package. This flavor is produced only if you specify
+    MLeap-compatible arguments.
+"""
+import json
+import logging
+import os
+import posixpath
+import shutil
+import sys
+from pathlib import Path
+from urllib.request import urlopen
+
+import johnsnowlabs.settings
+import yaml
+from johnsnowlabs import nlp
+from johnsnowlabs import settings
+from johnsnowlabs.auto_install.jsl_home import get_install_suite_from_jsl_home
+from johnsnowlabs.py_models.jsl_secrets import JslSecrets
+
+import mlflow
+from mlflow import pyfunc, mleap
+from mlflow.environment_variables import MLFLOW_DFS_TMP
+from mlflow.exceptions import MlflowException
+from mlflow.models import Model
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.models.signature import ModelSignature
+from mlflow.models.utils import ModelInputExample, _save_example
+from mlflow.spark import _mlflowdbfs_path, _should_use_mlflowdbfs, _maybe_save_model, _HadoopFileSystem
+from mlflow.store.artifact.databricks_artifact_repo import DatabricksArtifactRepository
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.tracking.artifact_utils import (
+    _download_artifact_from_uri,
+    _get_root_uri_and_artifact_path,
+)
+from mlflow.utils import databricks_utils
+from mlflow.utils.autologging_utils import autologging_integration, safe_patch
+from mlflow.utils.docstring_utils import format_docstring, LOG_MODEL_PARAM_DOCS
+from mlflow.utils.environment import (
+    _mlflow_conda_env,
+    _process_pip_requirements,
+    _process_conda_env,
+    _CONDA_ENV_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+    _CONSTRAINTS_FILE_NAME,
+    _PYTHON_ENV_FILE_NAME,
+    _PythonEnv,
+)
+from mlflow.utils.file_utils import TempDir, write_to, shutil_copytree_without_file_permissions
+from mlflow.utils.model_utils import (
+    _get_flavor_configuration_from_uri,
+    _validate_and_copy_code_paths,
+    _add_code_from_conf_to_system_path,
+)
+from mlflow.utils.uri import (
+    is_local_uri,
+    append_to_uri_path,
+    dbfs_hdfs_uri_to_fuse_path,
+    is_valid_dbfs_uri,
+    get_databricks_profile_uri_from_artifact_uri,
+    generate_tmp_dfs_path,
+)
+
+def validate_env_vars():
+    if not all(os.environ.get(var) for var in JOHNSNOWLABS_ENV_VARS):
+        raise Exception(
+            'Please set SECRET environment variable to your secret key before saving or logging your model')
```

Comment:
Can this exception be updated to call out which of the environment variable keys are missing? As it reads it would prompt a user to only validate that the SECRET key is set

Reply:
thanks, as suggested updated in latest commit